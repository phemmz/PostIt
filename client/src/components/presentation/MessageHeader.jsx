import React from 'react';
import PropTypes from 'prop-types';

/**
 * Home
 * @return {*} div
 */
const MessageHeader = ({ groupName }) => {
  return (
    <div className="row">
      <div className="col s12">
        <span><h5 id="welcome-header" className="green-text text-darken-4 card">
          <strong>
            { groupName }
          </strong></h5>
        </span>
      </div>
    </div>
  );
};

MessageHeader.propTypes = {
  groupName: PropTypes.string.isRequired
};

export default MessageHeader;
