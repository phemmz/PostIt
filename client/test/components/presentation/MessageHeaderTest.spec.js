import React from 'react';
import { mount } from 'enzyme';
import MessageHeader from '../../../src/components/presentation/MessageHeader';

describe('MessageHeader Component', () => {
  it('should render groupname of the group selected', () => {
    const wrapper = mount(<MessageHeader groupName="Random" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('groupName')).toBe('Random');
    expect(wrapper.find('div').exists()).toBe(true);
    expect(wrapper.find('span').exists()).toBe(true);
    expect(wrapper.find('h5').exists()).toBe(true);
    expect(wrapper.find('strong').exists()).toBe(true);
    expect(wrapper.find('div').length).toBe(2);
    expect(wrapper.find('span').length).toBe(1);
    expect(wrapper.find('h5').length).toBe(1);
    expect(wrapper.find('strong').length).toBe(1);
  });
});
