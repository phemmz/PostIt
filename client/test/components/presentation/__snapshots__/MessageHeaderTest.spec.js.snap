// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageHeader Component should render groupname of the group selected 1`] = `
<MessageHeader
  groupName="Random"
>
  <div
    className="row"
  >
    <div
      className="col s12"
    >
      <span>
        <h5
          className="green-text text-darken-4 card"
          id="welcome-header"
        >
          <strong>
            Random
          </strong>
        </h5>
      </span>
    </div>
  </div>
</MessageHeader>
`;
